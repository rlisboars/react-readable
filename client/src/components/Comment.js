import React, { Component } from 'react'
import { Segment, Grid, Button } from 'semantic-ui-react'
import { connect } from 'react-redux'
import { editComment, deleteContent, vote } from '../actions'

export class Comment extends Component {
    render() {
        const { data } = this.props
        const date = new Date(data.timestamp)
        return (
            <Segment>
                <Grid>
                    <Grid.Row>
                        <Grid.Column width={15}>
                            <span className='comment-user'>{data.author}</span>
                            <span className='comment-date'>{date.toLocaleString()}</span>
                        </Grid.Column>
                        <Grid.Column width={1}>
                            <Button.Group size='mini' basic floated='right'>
                                <Button icon='write' 
                                    onClick={(evt) => this.props.editComment(data.parentId, data.id)}
                                    compact />
                                <Button icon='trash' 
                                    onClick={(evt) => this.props.deleteContent(data.id, 'COMMENT')} 
                                    compact />
                            </Button.Group>
                        </Grid.Column>
                    </Grid.Row>
                    <Grid.Row className='comment-content'>
                        <Grid.Column>
                        {data.body}
                        </Grid.Column>
                    </Grid.Row>
                    <Grid.Row className='comment-content'>
                        <Grid.Column>
                            <Button 
                                icon='arrow up' 
                                size='mini' 
                                basic
                                style={{'marginRight':'0px'}} 
                                compact 
                                onClick={() => this.props.vote(data.id, 'COMMENT', 'upVote')}/>
                            <span className='comment-score'>{data.voteScore}</span>
                            <Button 
                                icon='arrow down' 
                                size='mini' 
                                basic 
                                compact
                                onClick={() => this.props.vote(data.id, 'COMMENT', 'downVote')}/>
                        </Grid.Column>
                    </Grid.Row>
                </Grid>
            </Segment>
        )
    }
}

export default connect(null, { editComment, deleteContent, vote })(Comment)