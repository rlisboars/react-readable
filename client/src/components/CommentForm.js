import React, { Component } from 'react'
import { Segment, Form } from 'semantic-ui-react'
import { connect } from 'react-redux'
import { setFilters, saveComment } from '../actions'

export class CommentForm extends Component {
    state = {
        body: '',
        author: ''
    }
    
    handleDisplay() {
        this.props.setFilters({
            ...this.props.filters,
            isCreatingComment: false
        })
    }

    handleFieldChange(field, value) {
        this.setState(currState => ({
            ...currState,
            [field]: value
        }))
    }

    saveComment() {
        const id = this.props.comment ? this.props.comment.id : undefined
        const comment = this.state
        if (!id) 
            this.props.saveComment({
                body: comment.body,
                author: comment.author,
                parentId: this.props.parentId
            }).then(comment => { 
                this.handleDisplay()
            })
        else this.props.saveComment({
            id: id,
            body: comment.body,
            parentId: this.props.parentId
        }).then(comment => {
            this.handleDisplay()
        })
    }

    componentWillMount() {
        const { comment } = this.props
        if (comment) {
            this.setState({
                body: comment.body,
                author: comment.author
            })
        }   
    }

    componentWillReceiveProps({comment}) {
        this.setState({
            body: comment.body,
            author: comment.author
        })
    }

    render() {
        const { comment } = this.props
        return(
            <Segment>
                <Form>
                    <Form.TextArea 
                        label='Body'
                        placeholder='Insert your comment'
                        name='body'
                        value={this.state.body}
                        onChange={(_,dt) => this.handleFieldChange(dt.name, dt.value)}
                        required />
                    <Form.Group>
                        <Form.Input 
                            label='Author'
                            placeholder='Your name'
                            width={3}
                            name='author'
                            value={this.state.author}
                            onChange={(_,dt) => this.handleFieldChange(dt.name, dt.value)}
                            disabled={comment ? true : false} 
                            required={!comment ? true: false} />
                        <Form.Field width={9} />
                        <Form.Button className='btn-form' width={2} fluid primary
                            onClick={(e) => { e.preventDefault(); this.saveComment()}}>Save</Form.Button>
                        <Form.Button className='btn-form' width={2} fluid 
                            onClick={(e) => { e.preventDefault(); this.handleDisplay()}}>Cancel</Form.Button>
                    </Form.Group>
                </Form>
            </Segment>
        )
    }
}

function mapStateToProps({ filters, changes, comments }, ownProps) {
    const { editingComment } = changes
    return editingComment 
        ? { filters, comment: comments[ownProps.parentId].find(cmt => cmt.id === editingComment)}
        : { filters, comment: null }
}

export default connect(mapStateToProps, { setFilters, saveComment })(CommentForm)