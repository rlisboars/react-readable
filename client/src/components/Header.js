import React, { Component } from 'react'
import { Menu, Select } from 'semantic-ui-react'
import { connect } from 'react-redux'
import { setFilters, getCategories } from '../actions'
import { Link } from 'react-router-dom'

export class Header extends Component {
	componentDidMount() {
		this.props.getCategories()
	}
	componentWillReceiveProps(newProps) {
		if (newProps.selected)
			if (this.props.filters.category !== newProps.selected.name) {
				this.props.setFilters({
					...this.props.filters,
					category: newProps.selected.name
				})
			}
	}
	handleSelectChange(selected) {
		this.props.setFilters({
			...this.props.filters,
			category: selected
		})
		this.props.history.push(`/${selected !== 'All' ? selected : ''}`)
	}
	render() {
		const { categories, disabled, filters } = this.props
		return (
		   <Menu borderless>
			   <Menu.Item>
				   <Link to='/' className='logo'>Leitura</Link>
			   </Menu.Item>
			   <Menu.Item>
				   <span className='category-select-label'>
					   Category:
				   </span>
				   <Select 
						options={
							categories.map((ct, idx) => ({'key': idx, 'value': ct.name, 'text': ct.name }))
						} 
						className='category-select' 
				   		placeholder='Choose a category' 
						disabled={disabled ? true : false} 
						value={filters.category}
						onChange={(evt, data) => this.handleSelectChange(data.value)}/>
			   </Menu.Item>
		   </Menu>
		)
	}
}

function mapStateToProps({ filters, categories }) {
	return { filters, categories }
}

export default connect(mapStateToProps, { setFilters, getCategories })(Header)