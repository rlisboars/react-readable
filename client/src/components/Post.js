import React, { Component } from 'react'
import { Menu, Segment, Grid, Header, Button } from 'semantic-ui-react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { editPost, vote, deleteContent } from '../actions'

export class Post extends Component {
    render() {
        const { data, category } = this.props
        let comments = this.props.comments 
                        ? this.props.comments.slice().filter(cmt => cmt.deleted === false)
                        :[]
        const date = new Date(data.timestamp)
        return (
            <Segment raised className='post'>
                <Grid divided stretched>
                    <Grid.Column width={2} textAlign='center' style={{ 'padding': '0' }}>
                        <Menu fluid vertical text>
                            <Menu.Item>
                                <Button icon='arrow up' 
                                    onClick={() => this.props.vote(data.id, 'POST', 'upVote')}
                                    compact />
                            </Menu.Item>
                            <Menu.Item>
                                <Header as='h4'>{data.voteScore}</Header>
                            </Menu.Item>
                            <Menu.Item>
                                <Button icon='arrow down' 
                                    onClick={() => this.props.vote(data.id, 'POST', 'downVote')}
                                    compact />
                            </Menu.Item>
                        </Menu>
                    </Grid.Column>
                    <Grid.Column width={14}>
                        <Grid>
                            <Grid.Row>
                                <Grid.Column width={14}>
                                    <Header as='h3'><Link to={`/${data.category}/${data.id}`}>{data.title}</Link></Header>
                                </Grid.Column>
                                <Grid.Column width={2} style={{ 'paddingLeft': '0px' }}>
                                    <Button.Group size='mini' basic floated='right'>
                                        <Button icon='write' compact
                                            onClick={(evt) => this.props.editPost(data.id)} />
                                        <Button icon='trash' compact
                                            onClick={(evt) => { 
                                                this.props.deleteContent(data.id, 'POST')
                                                if (this.props.history) this.props.history.push(`/${category !== 'All' ? category : ''}`)
                                            }} />
                                    </Button.Group>
                                </Grid.Column>
                            </Grid.Row>
                            { this.props.detailed && 
                                <Grid.Row style={{ 'paddingTop': '0px' }}>
                                    <Grid.Column>
                                        {data.body}
                                    </Grid.Column>
                                </Grid.Row>
                            }
                            <Grid.Row className='post-content-bottom'>
                                <Grid.Column width={12}>
                                    by {data.author} to {data.category} on { date.toLocaleString() }
                            </Grid.Column>
                                <Grid.Column width={4} textAlign='right'>
                                <Link to={`/${data.category}/${data.id}`}>{ comments ? comments.length : '0'} comments</Link>
                            </Grid.Column>
                            </Grid.Row>
                        </Grid>
                    </Grid.Column>
                </Grid>
            </Segment>
        )
    }
}

function mapStateToProps({ filters }) {
    return { category: filters.category }
}

export default connect(mapStateToProps, { editPost, vote, deleteContent })(Post)