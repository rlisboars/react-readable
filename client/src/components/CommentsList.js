import React, { Component } from 'react'
import { Menu, Segment, Message } from 'semantic-ui-react'
import HeaderReadable from './Header'
import Post from './Post'
import PostForm from './PostForm'
import Comment from './Comment'
import CommentForm from './CommentForm'
import ListActions from './ListActions'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { getPost } from '../actions'

export class CommentsList extends Component {
    componentDidMount() {
        this.props.getPost(this.props.match.params.id)
    }
    renderBackButton() {
        const { category } = this.props.filters
        return(
            <Menu className='sub-menu' text>
                <Menu.Item>
                    <Link to={ category === 'All' ? '/' : `/${category}`}>
                        Back
                    </Link>
                </Menu.Item>
            </Menu>
        )
    }
    render() {
        const { filters, post, changes } = this.props
        if (post && post.error) {
            return (
                <div>
                    <HeaderReadable history={this.props.history}/>
                    <Message negative className='infoMessage'>
                        <Message.Header>
                            {post.error}
                        </Message.Header>
                    </Message>
                </div>
            )
        }
        const sortBy = filters.sortByComment
        let postComments = this.props.postComments 
                            ? this.props.postComments.slice().filter(cmt => cmt.deleted === false) 
                            : []
        
        if (postComments) {
            switch (sortBy) {
                case 'recent':
                    postComments.sort((prev, curr) => curr.timestamp - prev.timestamp)
                    break
                case 'score':
                    postComments.sort((prev, curr) => curr.voteScore - prev.voteScore)
                    break
                default:
                    break
            }
        }

        const postId = this.props.match.params.id
        if (!post) {
            return (
                <div>
                    <HeaderReadable disabled />
                    {this.renderBackButton()}
                    <Message negative className='infoMessage'>
                        <Message.Header>Post id '{postId}' not found!</Message.Header>
                    </Message>
                </div>
            )
        } else {
            return (
                <div>
                    <HeaderReadable disabled />
                    {this.renderBackButton()}
                    { filters.isCreatingPost && post.id === changes.editingPost &&
                        <PostForm />
                    }
                    { !filters.isCreatingPost &&
                        <Post data={post} comments={postComments} detailed/>
                    }
                    <ListActions comment />
                    <Segment.Group>
                        {filters.isCreatingComment && !changes.editingComment &&
                            <CommentForm parentId={postId}/>
                        }
                        { postComments.length === 0 && !filters.isCreatingComment &&
                            <Message size='mini' color='teal' className='infoMessage'>No comments done yet! Be the first!</Message>
                        }
                        { postComments.length > 0 &&  
                            postComments.map(comment => {
                                if (filters.isCreatingComment && comment.id === changes.editingComment) {
                                    return <CommentForm key={comment.id} parentId={postId}/>
                                } else return <Comment data={comment} key={comment.id}/>
                            })                          
                        }
                    </Segment.Group>
                </div>
            )
        }
    }
}

function mapStateToProps({ filters, posts, comments, changes }, ownProps) {
    if (posts.error) return { filters, post: posts }
    const postsArray = Object.values(posts)
    const post = postsArray.length > 0 ? postsArray.find(p => p.id === ownProps.match.params.id) : undefined
    const postComments = post && comments[post.id] ? comments[post.id] : undefined
    return { filters, post, postComments, changes }
}

export default connect(mapStateToProps, { getPost })(CommentsList)