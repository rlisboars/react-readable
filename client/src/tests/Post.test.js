import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom'
import { Post } from '../components/Post'
import { mount, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-15'
import { store } from './mockStore'

const postData = { 
    author: 'thingtwo',
    body: 'Everyone says so after all.',
    category: 'react',
    deleted: false,
    id: "8xf0y6ziyjabvozdd253nd",
    timestamp: 1467166872634,
    title: "Udacity is the best place to learn React",
    voteScore: 6
}
describe('Post Component testing', () => {

    const voteFn = jest.fn()
    const deleteContentFn = jest.fn()
    const editPostFn = jest.fn()

    const postWithDetails = mount(
        <MemoryRouter>
            <Post 
                data={postData} 
                detailed='true' 
                vote={voteFn} 
                editPost={editPostFn} 
                deleteContent={deleteContentFn}/>
        </MemoryRouter>
    )
    const postWithoutDetails = mount(
        <MemoryRouter>
            <Post data={postData} />
        </MemoryRouter>
    )

    const postWithComment = mount(
        <MemoryRouter>
            <Post 
                data={postData} 
                comments={
                    [{
                        author: 'thingtwo',
                        body: 'Hi there! I am a COMMENT.',
                        deleted: false,
                        id: '894tuq4ut84ut8v4t8wun89g',
                        parentDeleted: false,
                        parentId: '8xf0y6ziyjabvozdd253nd',
                        timestamp: 1468166872634,
                        voteScore: 6
                    }]}/>
        </MemoryRouter>
    )

    it('##### renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <MemoryRouter>
                <Post data={postData} store={store}/>
            </MemoryRouter>
        , div);
    })

    it('#### validates snapshot of Post without details', () => {
        expect(toJson(postWithDetails.find(Post))).toMatchSnapshot()
    })

    it('#### validates snapshot of Post with details', () => {
        expect(toJson(postWithoutDetails.find(Post))).toMatchSnapshot()
    })

    it('### validates snapshot of Post with details and coments', () => {
        expect(toJson(postWithComment.find(Post))).toMatchSnapshot()
    })

    it('### vote up and vote down are called properly', () => {
        postWithDetails.find({ icon: 'arrow up' }).simulate('click')
        expect(voteFn).toHaveBeenCalledWith(postData.id, 'POST', 'upVote')
        postWithDetails.find({ icon: 'arrow down' }).simulate('click')
        expect(voteFn).toHaveBeenCalledWith(postData.id, 'POST', 'downVote')
    })

    it('### edit button is working properly', () => {
        postWithDetails.find({ icon: 'write' }).simulate('click')
        expect(editPostFn).toHaveBeenCalledWith(postData.id)
    })

    it('### delete button is working properly', () => {
        postWithDetails.find({ icon: 'trash' }).simulate('click')
        expect(deleteContentFn).toHaveBeenCalledWith(postData.id,'POST')
    })
})
