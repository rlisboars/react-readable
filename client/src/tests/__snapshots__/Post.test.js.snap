// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Component testing ### validates snapshot of Post with details and coments 1`] = `
<Post
  comments={
    Array [
      Object {
        "author": "thingtwo",
        "body": "Hi there! I am a COMMENT.",
        "deleted": false,
        "id": "894tuq4ut84ut8v4t8wun89g",
        "parentDeleted": false,
        "parentId": "8xf0y6ziyjabvozdd253nd",
        "timestamp": 1468166872634,
        "voteScore": 6,
      },
    ]
  }
  data={
    Object {
      "author": "thingtwo",
      "body": "Everyone says so after all.",
      "category": "react",
      "deleted": false,
      "id": "8xf0y6ziyjabvozdd253nd",
      "timestamp": 1467166872634,
      "title": "Udacity is the best place to learn React",
      "voteScore": 6,
    }
  }
>
  <Segment
    className="post"
    raised={true}
  >
    <div
      className="ui raised segment post"
    >
      <Grid
        divided={true}
        stretched={true}
      >
        <div
          className="ui stretched divided grid"
        >
          <GridColumn
            style={
              Object {
                "padding": "0",
              }
            }
            textAlign="center"
            width={2}
          >
            <div
              className="center aligned two wide column"
              style={
                Object {
                  "padding": "0",
                }
              }
            >
              <Menu
                fluid={true}
                text={true}
                vertical={true}
              >
                <div
                  className="ui fluid text vertical menu"
                >
                  <MenuItem>
                    <div
                      className="item"
                      onClick={[Function]}
                    >
                      <Button
                        as="button"
                        compact={true}
                        icon="arrow up"
                        onClick={[Function]}
                      >
                        <button
                          className="ui compact icon button"
                          disabled={undefined}
                          onClick={[Function]}
                          role="button"
                          tabIndex={undefined}
                        >
                          <Icon
                            as="i"
                            key="arrow up"
                            name="arrow up"
                          >
                            <i
                              aria-hidden="true"
                              className="arrow up icon"
                            />
                          </Icon>
                        </button>
                      </Button>
                    </div>
                  </MenuItem>
                  <MenuItem>
                    <div
                      className="item"
                      onClick={[Function]}
                    >
                      <Header
                        as="h4"
                      >
                        <h4
                          className="ui header"
                        >
                          6
                        </h4>
                      </Header>
                    </div>
                  </MenuItem>
                  <MenuItem>
                    <div
                      className="item"
                      onClick={[Function]}
                    >
                      <Button
                        as="button"
                        compact={true}
                        icon="arrow down"
                        onClick={[Function]}
                      >
                        <button
                          className="ui compact icon button"
                          disabled={undefined}
                          onClick={[Function]}
                          role="button"
                          tabIndex={undefined}
                        >
                          <Icon
                            as="i"
                            key="arrow down"
                            name="arrow down"
                          >
                            <i
                              aria-hidden="true"
                              className="arrow down icon"
                            />
                          </Icon>
                        </button>
                      </Button>
                    </div>
                  </MenuItem>
                </div>
              </Menu>
            </div>
          </GridColumn>
          <GridColumn
            width={14}
          >
            <div
              className="fourteen wide column"
            >
              <Grid>
                <div
                  className="ui grid"
                >
                  <GridRow>
                    <div
                      className="row"
                    >
                      <GridColumn
                        width={14}
                      >
                        <div
                          className="fourteen wide column"
                        >
                          <Header
                            as="h3"
                          >
                            <h3
                              className="ui header"
                            >
                              <Link
                                replace={false}
                                to="/post/8xf0y6ziyjabvozdd253nd"
                              >
                                <a
                                  href="/post/8xf0y6ziyjabvozdd253nd"
                                  onClick={[Function]}
                                >
                                  Udacity is the best place to learn React
                                </a>
                              </Link>
                            </h3>
                          </Header>
                        </div>
                      </GridColumn>
                      <GridColumn
                        style={
                          Object {
                            "paddingLeft": "0px",
                          }
                        }
                        width={2}
                      >
                        <div
                          className="two wide column"
                          style={
                            Object {
                              "paddingLeft": "0px",
                            }
                          }
                        >
                          <ButtonGroup
                            basic={true}
                            floated="right"
                            size="mini"
                          >
                            <div
                              className="ui mini basic right floated buttons"
                            >
                              <Button
                                as="button"
                                compact={true}
                                icon="write"
                                onClick={[Function]}
                              >
                                <button
                                  className="ui compact icon button"
                                  disabled={undefined}
                                  onClick={[Function]}
                                  role="button"
                                  tabIndex={undefined}
                                >
                                  <Icon
                                    as="i"
                                    key="write"
                                    name="write"
                                  >
                                    <i
                                      aria-hidden="true"
                                      className="write icon"
                                    />
                                  </Icon>
                                </button>
                              </Button>
                              <Button
                                as="button"
                                compact={true}
                                icon="trash"
                                onClick={[Function]}
                              >
                                <button
                                  className="ui compact icon button"
                                  disabled={undefined}
                                  onClick={[Function]}
                                  role="button"
                                  tabIndex={undefined}
                                >
                                  <Icon
                                    as="i"
                                    key="trash"
                                    name="trash"
                                  >
                                    <i
                                      aria-hidden="true"
                                      className="trash icon"
                                    />
                                  </Icon>
                                </button>
                              </Button>
                            </div>
                          </ButtonGroup>
                        </div>
                      </GridColumn>
                    </div>
                  </GridRow>
                  <GridRow
                    className="post-content-bottom"
                  >
                    <div
                      className="row post-content-bottom"
                    >
                      <GridColumn
                        width={12}
                      >
                        <div
                          className="twelve wide column"
                        >
                          by 
                          thingtwo
                           to 
                          react
                           on 
                          2016-06-28 23:21:12
                        </div>
                      </GridColumn>
                      <GridColumn
                        textAlign="right"
                        width={4}
                      >
                        <div
                          className="right aligned four wide column"
                        >
                          <Link
                            replace={false}
                            to="/post/8xf0y6ziyjabvozdd253nd"
                          >
                            <a
                              href="/post/8xf0y6ziyjabvozdd253nd"
                              onClick={[Function]}
                            >
                              1
                               comments
                            </a>
                          </Link>
                        </div>
                      </GridColumn>
                    </div>
                  </GridRow>
                </div>
              </Grid>
            </div>
          </GridColumn>
        </div>
      </Grid>
    </div>
  </Segment>
</Post>
`;

exports[`Post Component testing #### validates snapshot of Post with details 1`] = `
<Post
  data={
    Object {
      "author": "thingtwo",
      "body": "Everyone says so after all.",
      "category": "react",
      "deleted": false,
      "id": "8xf0y6ziyjabvozdd253nd",
      "timestamp": 1467166872634,
      "title": "Udacity is the best place to learn React",
      "voteScore": 6,
    }
  }
>
  <Segment
    className="post"
    raised={true}
  >
    <div
      className="ui raised segment post"
    >
      <Grid
        divided={true}
        stretched={true}
      >
        <div
          className="ui stretched divided grid"
        >
          <GridColumn
            style={
              Object {
                "padding": "0",
              }
            }
            textAlign="center"
            width={2}
          >
            <div
              className="center aligned two wide column"
              style={
                Object {
                  "padding": "0",
                }
              }
            >
              <Menu
                fluid={true}
                text={true}
                vertical={true}
              >
                <div
                  className="ui fluid text vertical menu"
                >
                  <MenuItem>
                    <div
                      className="item"
                      onClick={[Function]}
                    >
                      <Button
                        as="button"
                        compact={true}
                        icon="arrow up"
                        onClick={[Function]}
                      >
                        <button
                          className="ui compact icon button"
                          disabled={undefined}
                          onClick={[Function]}
                          role="button"
                          tabIndex={undefined}
                        >
                          <Icon
                            as="i"
                            key="arrow up"
                            name="arrow up"
                          >
                            <i
                              aria-hidden="true"
                              className="arrow up icon"
                            />
                          </Icon>
                        </button>
                      </Button>
                    </div>
                  </MenuItem>
                  <MenuItem>
                    <div
                      className="item"
                      onClick={[Function]}
                    >
                      <Header
                        as="h4"
                      >
                        <h4
                          className="ui header"
                        >
                          6
                        </h4>
                      </Header>
                    </div>
                  </MenuItem>
                  <MenuItem>
                    <div
                      className="item"
                      onClick={[Function]}
                    >
                      <Button
                        as="button"
                        compact={true}
                        icon="arrow down"
                        onClick={[Function]}
                      >
                        <button
                          className="ui compact icon button"
                          disabled={undefined}
                          onClick={[Function]}
                          role="button"
                          tabIndex={undefined}
                        >
                          <Icon
                            as="i"
                            key="arrow down"
                            name="arrow down"
                          >
                            <i
                              aria-hidden="true"
                              className="arrow down icon"
                            />
                          </Icon>
                        </button>
                      </Button>
                    </div>
                  </MenuItem>
                </div>
              </Menu>
            </div>
          </GridColumn>
          <GridColumn
            width={14}
          >
            <div
              className="fourteen wide column"
            >
              <Grid>
                <div
                  className="ui grid"
                >
                  <GridRow>
                    <div
                      className="row"
                    >
                      <GridColumn
                        width={14}
                      >
                        <div
                          className="fourteen wide column"
                        >
                          <Header
                            as="h3"
                          >
                            <h3
                              className="ui header"
                            >
                              <Link
                                replace={false}
                                to="/post/8xf0y6ziyjabvozdd253nd"
                              >
                                <a
                                  href="/post/8xf0y6ziyjabvozdd253nd"
                                  onClick={[Function]}
                                >
                                  Udacity is the best place to learn React
                                </a>
                              </Link>
                            </h3>
                          </Header>
                        </div>
                      </GridColumn>
                      <GridColumn
                        style={
                          Object {
                            "paddingLeft": "0px",
                          }
                        }
                        width={2}
                      >
                        <div
                          className="two wide column"
                          style={
                            Object {
                              "paddingLeft": "0px",
                            }
                          }
                        >
                          <ButtonGroup
                            basic={true}
                            floated="right"
                            size="mini"
                          >
                            <div
                              className="ui mini basic right floated buttons"
                            >
                              <Button
                                as="button"
                                compact={true}
                                icon="write"
                                onClick={[Function]}
                              >
                                <button
                                  className="ui compact icon button"
                                  disabled={undefined}
                                  onClick={[Function]}
                                  role="button"
                                  tabIndex={undefined}
                                >
                                  <Icon
                                    as="i"
                                    key="write"
                                    name="write"
                                  >
                                    <i
                                      aria-hidden="true"
                                      className="write icon"
                                    />
                                  </Icon>
                                </button>
                              </Button>
                              <Button
                                as="button"
                                compact={true}
                                icon="trash"
                                onClick={[Function]}
                              >
                                <button
                                  className="ui compact icon button"
                                  disabled={undefined}
                                  onClick={[Function]}
                                  role="button"
                                  tabIndex={undefined}
                                >
                                  <Icon
                                    as="i"
                                    key="trash"
                                    name="trash"
                                  >
                                    <i
                                      aria-hidden="true"
                                      className="trash icon"
                                    />
                                  </Icon>
                                </button>
                              </Button>
                            </div>
                          </ButtonGroup>
                        </div>
                      </GridColumn>
                    </div>
                  </GridRow>
                  <GridRow
                    className="post-content-bottom"
                  >
                    <div
                      className="row post-content-bottom"
                    >
                      <GridColumn
                        width={12}
                      >
                        <div
                          className="twelve wide column"
                        >
                          by 
                          thingtwo
                           to 
                          react
                           on 
                          2016-06-28 23:21:12
                        </div>
                      </GridColumn>
                      <GridColumn
                        textAlign="right"
                        width={4}
                      >
                        <div
                          className="right aligned four wide column"
                        >
                          <Link
                            replace={false}
                            to="/post/8xf0y6ziyjabvozdd253nd"
                          >
                            <a
                              href="/post/8xf0y6ziyjabvozdd253nd"
                              onClick={[Function]}
                            >
                              0
                               comments
                            </a>
                          </Link>
                        </div>
                      </GridColumn>
                    </div>
                  </GridRow>
                </div>
              </Grid>
            </div>
          </GridColumn>
        </div>
      </Grid>
    </div>
  </Segment>
</Post>
`;

exports[`Post Component testing #### validates snapshot of Post without details 1`] = `
<Post
  data={
    Object {
      "author": "thingtwo",
      "body": "Everyone says so after all.",
      "category": "react",
      "deleted": false,
      "id": "8xf0y6ziyjabvozdd253nd",
      "timestamp": 1467166872634,
      "title": "Udacity is the best place to learn React",
      "voteScore": 6,
    }
  }
  deleteContent={[Function]}
  detailed="true"
  editPost={[Function]}
  vote={[Function]}
>
  <Segment
    className="post"
    raised={true}
  >
    <div
      className="ui raised segment post"
    >
      <Grid
        divided={true}
        stretched={true}
      >
        <div
          className="ui stretched divided grid"
        >
          <GridColumn
            style={
              Object {
                "padding": "0",
              }
            }
            textAlign="center"
            width={2}
          >
            <div
              className="center aligned two wide column"
              style={
                Object {
                  "padding": "0",
                }
              }
            >
              <Menu
                fluid={true}
                text={true}
                vertical={true}
              >
                <div
                  className="ui fluid text vertical menu"
                >
                  <MenuItem>
                    <div
                      className="item"
                      onClick={[Function]}
                    >
                      <Button
                        as="button"
                        compact={true}
                        icon="arrow up"
                        onClick={[Function]}
                      >
                        <button
                          className="ui compact icon button"
                          disabled={undefined}
                          onClick={[Function]}
                          role="button"
                          tabIndex={undefined}
                        >
                          <Icon
                            as="i"
                            key="arrow up"
                            name="arrow up"
                          >
                            <i
                              aria-hidden="true"
                              className="arrow up icon"
                            />
                          </Icon>
                        </button>
                      </Button>
                    </div>
                  </MenuItem>
                  <MenuItem>
                    <div
                      className="item"
                      onClick={[Function]}
                    >
                      <Header
                        as="h4"
                      >
                        <h4
                          className="ui header"
                        >
                          6
                        </h4>
                      </Header>
                    </div>
                  </MenuItem>
                  <MenuItem>
                    <div
                      className="item"
                      onClick={[Function]}
                    >
                      <Button
                        as="button"
                        compact={true}
                        icon="arrow down"
                        onClick={[Function]}
                      >
                        <button
                          className="ui compact icon button"
                          disabled={undefined}
                          onClick={[Function]}
                          role="button"
                          tabIndex={undefined}
                        >
                          <Icon
                            as="i"
                            key="arrow down"
                            name="arrow down"
                          >
                            <i
                              aria-hidden="true"
                              className="arrow down icon"
                            />
                          </Icon>
                        </button>
                      </Button>
                    </div>
                  </MenuItem>
                </div>
              </Menu>
            </div>
          </GridColumn>
          <GridColumn
            width={14}
          >
            <div
              className="fourteen wide column"
            >
              <Grid>
                <div
                  className="ui grid"
                >
                  <GridRow>
                    <div
                      className="row"
                    >
                      <GridColumn
                        width={14}
                      >
                        <div
                          className="fourteen wide column"
                        >
                          <Header
                            as="h3"
                          >
                            <h3
                              className="ui header"
                            >
                              <Link
                                replace={false}
                                to="/post/8xf0y6ziyjabvozdd253nd"
                              >
                                <a
                                  href="/post/8xf0y6ziyjabvozdd253nd"
                                  onClick={[Function]}
                                >
                                  Udacity is the best place to learn React
                                </a>
                              </Link>
                            </h3>
                          </Header>
                        </div>
                      </GridColumn>
                      <GridColumn
                        style={
                          Object {
                            "paddingLeft": "0px",
                          }
                        }
                        width={2}
                      >
                        <div
                          className="two wide column"
                          style={
                            Object {
                              "paddingLeft": "0px",
                            }
                          }
                        >
                          <ButtonGroup
                            basic={true}
                            floated="right"
                            size="mini"
                          >
                            <div
                              className="ui mini basic right floated buttons"
                            >
                              <Button
                                as="button"
                                compact={true}
                                icon="write"
                                onClick={[Function]}
                              >
                                <button
                                  className="ui compact icon button"
                                  disabled={undefined}
                                  onClick={[Function]}
                                  role="button"
                                  tabIndex={undefined}
                                >
                                  <Icon
                                    as="i"
                                    key="write"
                                    name="write"
                                  >
                                    <i
                                      aria-hidden="true"
                                      className="write icon"
                                    />
                                  </Icon>
                                </button>
                              </Button>
                              <Button
                                as="button"
                                compact={true}
                                icon="trash"
                                onClick={[Function]}
                              >
                                <button
                                  className="ui compact icon button"
                                  disabled={undefined}
                                  onClick={[Function]}
                                  role="button"
                                  tabIndex={undefined}
                                >
                                  <Icon
                                    as="i"
                                    key="trash"
                                    name="trash"
                                  >
                                    <i
                                      aria-hidden="true"
                                      className="trash icon"
                                    />
                                  </Icon>
                                </button>
                              </Button>
                            </div>
                          </ButtonGroup>
                        </div>
                      </GridColumn>
                    </div>
                  </GridRow>
                  <GridRow
                    style={
                      Object {
                        "paddingTop": "0px",
                      }
                    }
                  >
                    <div
                      className="row"
                      style={
                        Object {
                          "paddingTop": "0px",
                        }
                      }
                    >
                      <GridColumn>
                        <div
                          className="column"
                        >
                          Everyone says so after all.
                        </div>
                      </GridColumn>
                    </div>
                  </GridRow>
                  <GridRow
                    className="post-content-bottom"
                  >
                    <div
                      className="row post-content-bottom"
                    >
                      <GridColumn
                        width={12}
                      >
                        <div
                          className="twelve wide column"
                        >
                          by 
                          thingtwo
                           to 
                          react
                           on 
                          2016-06-28 23:21:12
                        </div>
                      </GridColumn>
                      <GridColumn
                        textAlign="right"
                        width={4}
                      >
                        <div
                          className="right aligned four wide column"
                        >
                          <Link
                            replace={false}
                            to="/post/8xf0y6ziyjabvozdd253nd"
                          >
                            <a
                              href="/post/8xf0y6ziyjabvozdd253nd"
                              onClick={[Function]}
                            >
                              0
                               comments
                            </a>
                          </Link>
                        </div>
                      </GridColumn>
                    </div>
                  </GridRow>
                </div>
              </Grid>
            </div>
          </GridColumn>
        </div>
      </Grid>
    </div>
  </Segment>
</Post>
`;
